### PR TITLE
Html for curated content

### LINKED ISSUE #5472: Frontend calls DCR to get `related-stories` html
DCR exposes the `/Onwards` endpoint which accepts

```ts
type CAPIOnwardsType = {
	heading: string;
	trails: CAPITrailType[];
	description?: string;
	url?: string;
	onwardsType: OnwardsType;
	format: CAPIFormat;
	isCuratedContent?: boolean;
};
```

and returns the html for this component

When an api call from the client is made to fetch `related-stories` and if that html call contains the `html=true` query param, replace the json that we currently send as the response with `{ html: theResponseFromDCROnwards }`

See similar work: https://github.com/guardian/frontend/pull/25284

### PR DESCRIPTION
**Pull request by alinaboghiu**
## What does this change?

Adds the option for curated content to be rendered by DCR when an "html=true" parameter is present. 

This will be returned as json with an html field: 

```
{
  "html": "...."
}
```

Fixes https://github.com/guardian/dotcom-rendering/issues/5472

### CODE REVIEW

**Comment by oliverlloyd** [comment]
This fixes https://github.com/guardian/dotcom-rendering/issues/5472

File: common/app/renderers/DotcomRenderingService.scala [hunk 1]
```diff
@@ -31,6 +25,7 @@ import play.api.libs.ws.{WSClient, WSResponse}
 import play.api.mvc.Results.{InternalServerError, NotFound}
 import play.api.mvc.{RequestHeader, Result}
 import play.twirl.api.Html
+import services.NewsletterData
```
Comment (oliverlloyd) [inline common/app/renderers/DotcomRenderingService.scala:28]: Is this an IDE thing?

Comment (alinaboghiu) [reply common/app/renderers/DotcomRenderingService.scala:28]: yep Intellij just rearranged the imports

File: common/app/model/dotcomrendering/DotcomOnwardsRenderingDataModel.scala [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+// -----------------------------------------------------------------
+// DCR Onwards DataModel
+// -----------------------------------------------------------------
+
+case class DotcomOnwardsRenderingDataModel(
+    heading: String,
+    trails: Seq[OnwardItem],
+    ophanComponentName: String,
+    isCuratedContent: Boolean,
+)
```
Comment (oliverlloyd) [inline common/app/model/dotcomrendering/DotcomOnwardsRenderingDataModel.scala:16]: We need to add `format` here. The latest DCR version of this model now expects

```ts
type CAPIOnwardsType = {
	heading: string;
	trails: CAPITrailType[];
	ophanComponentName: OphanComponentName;
	format: CAPIFormat;
	isCuratedContent?: boolean;
};
```

where

```ts
type CAPIFormat = {
	design: CAPIDesign; // string
	theme: CAPITheme; // string
	display: CAPIDisplay; // string
};
```

Comment (alinaboghiu) [reply common/app/model/dotcomrendering/DotcomOnwardsRenderingDataModel.scala:16]: Format is available on each "trail" (CAPITrailType). If we do want to add format at this level, what format would make sense on a collection of trails?

Comment (oliverlloyd) [reply common/app/model/dotcomrendering/DotcomOnwardsRenderingDataModel.scala:16]: Hmm, this is more complex than I thought. We need to know the format to decide colouring for the container title and dot
Screenshot 2022-07-27 at 15 23 56


I initially thought we'd have the article id in the request. Most onwards calls do. 

Related example
https://code.api.nextgen.guardianapps.co.uk/related/business/2022/jul/27/china-property-sales-could-plunge-by-one-third-analysts-say-as-crisis-deepens.json?dcr=true

But for curated content that isn't the case so there's no way to know which format to use 🤔 

Curated example
https://code.api.nextgen.guardianapps.co.uk/container/data/754c-8e8c-fad9-a927.json?html

**Pull request closed without merging**